src/internal/standalone-types-check.c

```diff
@@ -17,3 +27,136 @@ r_obj* ffi_standalone_is_bool(r_obj* x,
 
   return r_true;
 }
+
+r_obj* ffi_standalone_check_number(r_obj* x,
+                                   r_obj* allow_decimal,
+                                   r_obj* min,
+                                   r_obj* max,
+                                   r_obj* allow_infinite,
+                                   r_obj* allow_na,
+                                   r_obj* allow_null) {
+  int out = IS_NUMBER_false;
+
+  switch (r_typeof(x)) {
+  case R_TYPE_null:
+    out = r_as_bool(allow_null) ? IS_NUMBER_true : IS_NUMBER_false;
+    break;
+
+  case R_TYPE_logical:
+    if (r_length(x) == 1 && r_lgl_get(x, 0) == r_globals.na_lgl) {
+      out = r_as_bool(allow_na) ? IS_NUMBER_true : IS_NUMBER_false;
+    }
+    break;
+
+  case R_TYPE_integer:
+    out = int_standalone_check_number(x,
+                                      min,
+                                      max,
+                                      allow_na,
+                                      allow_null);
+    break;
+
+  case R_TYPE_double:
+    out = dbl_standalone_check_number(x,
+                                      allow_decimal,
+                                      min,
+                                      max,
+                                      allow_infinite,
+                                      allow_na,
+                                      allow_null);
+    break;
+
+  default:
+    break;
+  }
+
+  return r_int(out);
+}
+
+static
+bool is_numeric(r_obj* x) {
+  if (r_attrib_get(x, r_syms.class_) == r_null) {
```

Comment:
```suggestion
  if (r_class(x) == r_null) {
```

?

Also did you just want `r_is_object()`?

Reply:
oh yeah good point I forgot about is-object